src/Logary/Scheduling.fs

```diff
@@ -4,74 +4,90 @@ module Logary.Internals.Scheduling
 
 // creds to Dave Thomas for his F# snippet
 open System.Threading
-  
-open FSharp.Actor
+
+open Hopac
+open Hopac.Infixes
 
 open NodaTime
 
+type Cancellation = private { cancelled: IVar<unit> }
```

Comment:
Yes, this will work, but I would suggest a slightly different design.  Specifically, only expose cancellation as a `Promise<unit>`.  This allow plugging in a nack (from withNack) as the cancellation token.